src/librustc_trans/trans/_match.rs

```diff
@@ -272,7 +272,10 @@ impl<'a, 'tcx> Opt<'a, 'tcx> {
         }
     }
 
-    fn trans<'blk>(&self, mut bcx: Block<'blk, 'tcx>) -> OptResult<'blk, 'tcx> {
+    fn trans<'r, 'blk>(&self,
+                       &mut Block { bl, ref mut fcx }: &mut Block<'r, 'blk, 'tcx>)
+                       -> OptResult<'blk> {
+        let mut bcx = &mut bl.with(fcx);
```

Comment:
This is a bit weird - especially to newcomers to trans - we take a Block as parameter, then create a new one, using only the components of the passed Block, then later in the function we use the fields of the same Block again. Either `with` needs a better name indicating what is going on, or we need a comment here.